buildSrc/src/main/java/org/elasticsearch/gradle/ConcatFilesTask.java

```
@@ -0,0 +1,93 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+ package org.elasticsearch.gradle;
+
+ import org.gradle.api.DefaultTask;
+ import org.gradle.api.file.FileTree;
+ import org.gradle.api.tasks.Input;
+ import org.gradle.api.tasks.InputFiles;
+ import org.gradle.api.tasks.Optional;
+ import org.gradle.api.tasks.OutputFile;
+ import org.gradle.api.tasks.TaskAction;
+
+ import java.io.*;
+ import java.nio.charset.StandardCharsets;
+ import java.util.Arrays;
+ import java.util.LinkedHashSet;
+ import java.util.Scanner;
+
+/**
+ * Concatenates a list of files into one and removes duplicate lines.
+ */
+public class ConcatFilesTask extends DefaultTask {
+
+    /** List of files to concatenate */
+    @InputFiles
+    public FileTree files;
+
+    /** line to add at the top of the target file */
+    @Input
+    @Optional
+    public String headerLine;
+
+    @OutputFile
+    public File target;
+
+    public ConcatFilesTask() {
+        final String description = "Concat a list of files into one.";
+    }
+
+    @TaskAction
+    public void concatFiles() {
+        final StringBuilder output = new StringBuilder();
+
+        if (!headerLine.equals("")) {
+            output.append(headerLine).append('\n');
+        }
+
+        final StringBuilder sb = new StringBuilder();
+        for (File f: files) {
+            try{
+                Scanner scanner = new Scanner(f, "UTF-8" );
+                String text = scanner.useDelimiter("\\A").next();
+                scanner.close();
+                sb.append(text);
+            } catch (FileNotFoundException e) {
+                e.printStackTrace();
+            }
+        }
+
+        // Remove duplicate lines
+        LinkedHashSet<String> uniqueLines = new LinkedHashSet<>(Arrays.asList(sb.toString().split("\n")));
+        try {
+            OutputStreamWriter osw = new OutputStreamWriter(new FileOutputStream(target), StandardCharsets.UTF_8);
```

After deduplicating in a List<String>, we can use `Files.write`. No need for hiding exceptions or dealing with closing/flushing. This `concatFiles()` method can have a checked IOException.